integration/presto/src/main/java/org/apache/carbondata/presto/ColumnarVectorWrapperDirect.java

```diff
@@ -0,0 +1,320 @@
+    this.columnVector = columnVector;
+    this.dictionaryVector = columnVector.getDictionaryVector();
+    this.nullBitset = new BitSet();
+  }
+
+  @Override public void setNullBits(BitSet nullBits) {
+    this.nullBitset = nullBits;
+  }
+
+  @Override public void putBoolean(int rowId, boolean value) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putBoolean(rowId, value);
+    }
+  }
+
+  @Override public void putFloat(int rowId, float value) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putFloat(rowId, value);
+    }
+  }
+
+  @Override public void putShort(int rowId, short value) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putShort(rowId, value);
+    }
+  }
+
+  @Override public void putShorts(int rowId, int count, short value) {
+    for (int i = 0; i < count; i++) {
+      if (nullBitset.get(rowId)) {
+        columnVector.putNull(rowId);
+      } else {
+        columnVector.putShort(rowId, value);
+      }
+      rowId++;
+    }
+
+  }
+
+  @Override public void putInt(int rowId, int value) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putInt(rowId, value);
+    }
+  }
+
+  @Override public void putInts(int rowId, int count, int value) {
+    columnVector.putInts(rowId, count, value);
+  }
+
+  @Override public void putLong(int rowId, long value) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putLong(rowId, value);
+    }
+  }
+
+  @Override public void putLongs(int rowId, int count, long value) {
+    columnVector.putLongs(rowId, count, value);
+  }
+
+  @Override public void putDecimal(int rowId, BigDecimal value, int precision) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putDecimal(rowId, value, precision);
+    }
+  }
+
+  @Override public void putDecimals(int rowId, int count, BigDecimal value, int precision) {
+    for (int i = 0; i < count; i++) {
+      if (nullBitset.get(rowId)) {
+        columnVector.putNull(rowId);
+      } else {
+        columnVector.putDecimal(rowId, value, precision);
+      }
+      rowId++;
+    }
+  }
+
+  @Override public void putDouble(int rowId, double value) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putDouble(rowId, value);
+    }
+  }
+
+  @Override public void putDoubles(int rowId, int count, double value) {
+    columnVector.putDoubles(rowId, count, value);
+  }
+
+  @Override public void putByteArray(int rowId, byte[] value) {
+    if (nullBitset.get(rowId)) {
+      columnVector.putNull(rowId);
+    } else {
+      columnVector.putByteArray(rowId, value);
+    }
+  }
+
+  @Override
+  public void putBytes(int rowId, int count, byte[] value) {
```

Comment:
It seems this is only used for filling default value, can we change its signature and avoid the null bit check inside?

Reply:
ok, removed the check and renamed